### ISSUE TITLE
Decorations

### ISSUE THREAD

**Issue by Theheffaboy** [BEEmod/BEE2-items#4361]
### Description of the feature

-add props to make maps look more lively (e.g:broken frames for overgrown, support pillars for old aperture,)
-add custom skyboxs (e.g:A 3d skybox of an overgrown facility,rubble from the destruction of GLaDOS in old aperture)
-add custom vital apparatus vents (e.g make the player fall into one and it opens to a completely different test chamber)
-add pipes from the Aperture Science Pipe Network and make it like the catwalks, and make it so it automatically spawns props for it to go down, also let there be an option for it to be enabled for like overgrown maps, also make it so they can also connect to the vital apparatus vents (like if there's an exposed part of a dropper, when it needs a cube, it can connect to it so when a cube is dropped, another cube will come through and wait for it to be dropped)
-allow the player to make custom rooms (e.g: custom observation room, custom entry/exit corridor,)
-add cores/custom cores (e.g:a core that guides the player through the map, wheatley watching over you)
-add custom textures for walls ,floors ,ceilings etc. (e.g: making a hallway inside a factory, allow the player to change the wall to a factory wall texture)

### Why should this be implemented

if this was implemented, it would make it easier for people that don't know how to use hammer editor or are having problems with hammer to edit their maps, and it would help for certain maps like lets say you escaped a test chamber and are going through offices, it would make it easier for the player to make said offices, if they want to have a custom intro lets say, going through an chamber lock in the offices to get to the test chamber, they could do that, or if a test is to hard, they can have someone to guide the player, or your map is outside offices and test chambers, the skyboxs would make it look more like a portal 2 map, or like if you want the player to be sucked up and sent back to the test chambers. you can put them in the pipe and they get dropped off at a vital apparatus vent. and to be honest, the props and such will make it look more like a portal map, and would make it more easier for people who don't know how to use hammer, or people who's hammer is broken (like mine).

### Additional information

here are some images that can demonstrate some of it 
![Untitled-1](https://github.com/BEEmod/BEE2-items/assets/97221998/9e075d8a-40ba-4d80-94fb-6d697b5bf5c3)
![Untitled](https://github.com/BEEmod/BEE2-items/assets/97221998/26c6d17b-aa79-4058-a183-016fa7fb6c29)
![Untitled-1](https://github.com/BEEmod/BEE2-items/assets/97221998/77121493-12ae-45fe-b312-a7386c57616c)
![images](https://github.com/BEEmod/BEE2-items/assets/97221998/1e10ccc4-758e-4740-be83-bd06be3f5c38)

**Comment by HeimlichLaboratories** [comment]
Not a dev, but:

-Decorations have already been suggested A LOT and iirc they're being worked on as auto generated instances.

-They're not gonna add cores, office rooms, or otherwise anything not in a test chamber. BEE2 is not an easy Hammer, it's only for puzzles. Learn Hammer if you want to make custom rooms.

**Comment by HeimlichLaboratories** [comment]
Also, how is your hammer broken? what's wrong?

**Comment by Theheffaboy** [comment]
1: i mean like manually place props, and plus, Isn't the point of BEE2 to extend the portal 2 puzzle editor's abilities,
2: its broken as in i cant compile and run maps when they're exported from the puzzle editor when they're made with bee2 because some of the assets like elevator corridors are missing

**Comment by FrozenDragon0** [comment]
> 1: i mean like manually place props, and plus, Isn't the point of BEE2 to extend the portal 2 puzzle editor's abilities, 2: its broken as in i cant compile and run maps when they're exported from the puzzle editor when they're made with bee2 because some of the assets like elevator corridors are missing

1. There's no point in manually placing props once the decorations system is put in place.
2. Do not export puzzlemaker maps into hammer as that is bad practice and the map itself will just be a pain to work with due to the way puzzlemaker maps are made in general.

**Comment by HeimlichLaboratories** [comment]
> 1: i mean like manually place props, and plus, Isn't the point of BEE2 to extend the portal 2 puzzle editor's abilities, 2: its broken as in i cant compile and run maps when they're exported from the puzzle editor when they're made with bee2 because some of the assets like elevator corridors are missing

Yes, to extend the PUZZLE editor's abilities. To extend the PUZZLE creation abilities. It isn't a tool for little portal timmies to easily create awful looking BTS areas.

**Comment by ctrlcmdmaster** [comment]
> > 1: i mean like manually place props, and plus, Isn't the point of BEE2 to extend the portal 2 puzzle editor's abilities, 2: its broken as in i cant compile and run maps when they're exported from the puzzle editor when they're made with bee2 because some of the assets like elevator corridors are missing
> 
> Yes, to extend the PUZZLE editor's abilities. To extend the PUZZLE creation abilities. It isn't a tool for little portal timmies to easily create awful looking BTS areas.

Okay, look, I may have been completely unrelated to this conversation before, and I know it's old, so I don't mean to butt into the conversation or be mean, but STILL! THESE FEATURES NEED TO BE ADDED, not just for "little portal timmies", but for people who don't know how to use hammer, and actually want to make cool, awesome maps, with cool cores, or good-looking exits, THIS IS ESSENTIAL to the future BEEmod experience, and sure, I may just sound like an angry nerd, but all of us in the portal community are when you think about it! Again, sorry for the rant (if anybody even sees this), but PLEASE, PLEASE add these features.

**Comment by HeimlichLaboratories** [comment]
> > > 1: i mean like manually place props, and plus, Isn't the point of BEE2 to extend the portal 2 puzzle editor's abilities, 2: its broken as in i cant compile and run maps when they're exported from the puzzle editor when they're made with bee2 because some of the assets like elevator corridors are missing
> > 
> > 
> > Yes, to extend the PUZZLE editor's abilities. To extend the PUZZLE creation abilities. It isn't a tool for little portal timmies to easily create awful looking BTS areas.
> 
> Okay, look, I may have been completely unrelated to this conversation before, and I know it's old, so I don't mean to butt into the conversation or be mean, but STILL! THESE FEATURES NEED TO BE ADDED, not just for "little portal timmies", but for people who don't know how to use hammer, and actually want to make cool, awesome maps, with cool cores, or good-looking exits, THIS IS ESSENTIAL to the future BEEmod experience, and sure, I may just sound like an angry nerd, but all of us in the portal community are when you think about it! Again, sorry for the rant (if anybody even sees this), but PLEASE, PLEASE add these features.

Well damn, didn't think I'd ever go back to this account, but I got the email, so I guess here I am.

Look. "not knowing how to use hammer" is not a disability beyond your control. If you want to make awesome maps with stories and personality cores, learn to use it. I did, and it's really not that hard. As it stands, BEEmod is strictly a mod for the puzzlemaker. Sure, it offers some basic decoration features, but essentially, it's still the puzzlemaker. I don't think you understand how difficult these features would be to implement either - a lot of the puzzlemaker is hardcoded and the devs have to "dance" around the restrictions or make use of existing features to implement others.

It is not essential to the future BEEmod experience. They are not adding story-creation features to BEEmod.

**Comment by LincoCompany00** [comment]
I mean yeah. BEEmod is intended specifically for puzzles. I think there might be an automatic decoration system being worked on, but adding decorations either way is going to be much harder than it seems. I think. I mean, I'm not a developer, so I'm just guessing. 

Also, having decorations to allow you to make storylines would be extremely complicated in an editor like the PTI in-game editor, because to make a unique story you'd have to have 55 bajilion different items for every decoration. Not to mention Hammer allows much more customization. Hammer may look daunting, but there's a surprising amount you can do if you know the basics. There's lots of tutorial videos on YouTube on using Hammer, not to mention you'll learn a lot more as you use it.

**Comment by ctrlcmdmaster** [comment]
Fine then guys. I suppose you have a good point, maybe I'll learn hammer++.

**Comment by HeimlichLaboratories** [comment]
> Fine then guys. I suppose you have a good point, maybe I'll learn hammer++.

Welcome to the dark side! /s
